src/Extensions/SiteTreeSubsites.php

```diff
@@ -382,17 +382,16 @@ public function alternateAbsoluteLink($action = null)
 
     /**
      * Use the CMS domain for iframed CMS previews to prevent single-origin violations
-     * and SSL cert problems.
+     * and SSL cert problems. Always set SubsiteID to avoid errors because a page doesn't
+     * exist on the CMS domain.
      * @param null $action
      * @return string
      */
-    public function alternatePreviewLink($action = null)
```

Comment:
This would be a breaking change. How much do you care about it? Looks like you need to change the arguments too. You could either do that in a non breaking way (using `func_get_args`) or create a new method and deprecate the old method. Would be good to get this out without a major version bump though!

Reply:
Hmm, a good (overlooked) point. Will fix.
The reasoning for the original change is [deprecation](https://github.com/silverstripe/silverstripe-cms/blob/4.0.0/code/Model/SiteTree.php#L560)

Reply:
I see your deprecation and raise you with... more deprecation